account_cutoff_prepaid/models/account_cutoff.py

```diff
@@ -0,0 +1,180 @@
+            prepaid_days = total_days - out_days
+        else:
+            cutoff_date_str = self.cutoff_date
+            cutoff_date_dt = fields.Date.from_string(cutoff_date_str)
+            if start_date_dt > cutoff_date_dt:
+                prepaid_days = total_days
+            else:
+                prepaid_days = (end_date_dt - cutoff_date_dt).days
+        assert total_days > 0,\
+            'Should never happen. Total days should always be > 0'
+        cutoff_amount = (aml.debit - aml.credit) *\
+            prepaid_days / float(total_days)
+        # we use account mapping here
+        if aml.account_id.id in mapping:
+            cutoff_account_id = mapping[aml.account_id.id]
+        else:
+            cutoff_account_id = aml.account_id.id
+
+        res = {
+            'parent_id': self.id,
+            'move_line_id': aml.id,
+            'partner_id': aml.partner_id.id or False,
+            'name': aml.name,
+            'start_date': start_date_dt,
+            'end_date': end_date_dt,
+            'account_id': aml.account_id.id,
+            'cutoff_account_id': cutoff_account_id,
+            'analytic_account_id': aml.analytic_account_id.id or False,
+            'total_days': total_days,
+            'prepaid_days': prepaid_days,
+            'amount': aml.credit - aml.debit,
+            'currency_id': self.company_currency_id.id,
+            'cutoff_amount': cutoff_amount,
+            }
+        return res
+
+    @api.multi
+    def get_prepaid_lines(self):
+        self.ensure_one()
+        aml_obj = self.env['account.move.line']
+        line_obj = self.env['account.cutoff.line']
+        mapping_obj = self.env['account.cutoff.mapping']
+        if not self.source_journal_ids:
+            raise UserError(
+                _("You should set at least one Source Journal."))
+        cutoff_date_str = self.cutoff_date
+        # Delete existing lines
+        self.line_ids.unlink()
+
+        if self.forecast:
+            domain = [
+                ('start_date', '<=', self.end_date),
+                ('end_date', '>=', self.start_date),
+                ('journal_id', 'in', self.source_journal_ids.ids)
+                ]
+        else:
+            domain = [
+                ('start_date', '!=', False),
+                ('journal_id', 'in', self.source_journal_ids.ids),
+                ('end_date', '>', cutoff_date_str),
+                ('date', '<=', cutoff_date_str)
+                ]
+
+        # Search for account move lines in the source journals
+        amls = aml_obj.search(domain)
+        # Create mapping dict
+        mapping = mapping_obj._get_mapping_dict(self.company_id.id, self.type)
+
+        # Loop on selected account move lines to create the cutoff lines
+        for aml in amls:
+            line_obj.create(self._prepare_prepaid_lines(aml, mapping))
+        return True
+
+    @api.model
+    def _inherit_default_cutoff_account_id(self):
```

Comment:
if you proxy the method in the field (as stated in the base PR) you have to remove `_inherit_` here